From 377d421c28d9cc99b7c463ff65a4481e4972f4e9 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sun, 30 Mar 2014 15:01:55 +0200
Subject: [PATCH 8/8] alsa-extcon: don't use is_input

* is_input was removed in:
  commit 0bbef56976496084bf04cee703a6b7e62d457cb3
  Author: Tanu Kaskinen <tanu.kaskinen@digia.com>
  Date:   Fri Jun 29 18:04:59 2012 +0300

    device-port: Make it impossible to have dual-direction ports

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/modules/alsa/alsa-extcon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/alsa/alsa-extcon.c b/src/modules/alsa/alsa-extcon.c
index 87b3616..663b41f 100644
--- a/src/modules/alsa/alsa-extcon.c
+++ b/src/modules/alsa/alsa-extcon.c
@@ -128,7 +128,7 @@ static void notify_ports(pa_alsa_extcon *u, struct android_switch *as) {
     pa_log_debug("Value of switch %s is now %d.", as->name, as->current_value);
 
     PA_HASHMAP_FOREACH(p, u->card->ports, state) {
-        if (p->is_output) {
+	if (p->direction == PA_ALSA_DIRECTION_OUTPUT) {
             if (!strcmp(p->name, "analog-output-headphones"))
                  pa_device_port_set_available(p, hp_avail(as->current_value));
 /* IFDEF HAVE_UCM ? */
@@ -137,7 +137,7 @@ static void notify_ports(pa_alsa_extcon *u, struct android_switch *as) {
                 pa_device_port_set_available(p, hp_avail(as->current_value));
 /* ENDIF */
         }
-        if (p->is_input) {
+        if (p->direction == PA_ALSA_DIRECTION_INPUT) {
             if (!strcmp(p->name, "analog-input-headset-mic"))
                 pa_device_port_set_available(p, hsmic_avail(as->current_value));
 /* IFDEF HAVE_UCM ? */
-- 
1.9.1

